.circleci/config.yml

```diff
@@ -70,236 +76,295 @@ parameters:
 common_job_environment: &common_job_environment
   CARGO_NET_GIT_FETCH_WITH_CLI: true
   RUST_BACKTRACE: full
-
+  CARGO_INCREMENTAL: 0
+  RUST_TEST_THREADS: 6
```

Comment:
Previously, this was fine-tuned by platform, because some platforms could handle more test threads than others. I think this change unifies it across all platforms, so does that mean we have some (e.g.: arm) running slower than they could and thus holding up the build?

Reply:
Hmmm, I think that this fine tuning should be moved into xtask.

Currently linux users need to know and appropriately set these setting to not have their desktops fall apart when running cargo test. I spent a few months watching the output of the terminal to try and stop the thing before it hit the docs stage of the test, but if I failed then all my running processes would get terminated.

WDYT?